src/components/noticeboard/NoticeboardCreateForm.tsx

```diff
@@ -44,35 +53,56 @@ const NOTICEBOARD_TYPE_OPTIONS = [
   }
 ];
 
+/* eslint-disable complexity */
 export const NoticeboardCreateForm = ({
+  data,
   navigation,
   route
 }: {
+  data: any;
   navigation: StackNavigationProp<any>;
   route: any;
 }) => {
-  const consentForDataProcessingText = route?.params?.consentForDataProcessingText ?? '';
+  const isEdit = !!Object.keys(data).length;
+  const subQuery = route.params?.subQuery ?? {};
+  const consentForDataProcessingText =
+    subQuery?.params?.consentForDataProcessingText ??
+    route?.params?.consentForDataProcessingText ??
+    '';
   const genericType = route?.params?.genericType ?? '';
   const requestedDateDifference = route?.params?.requestedDateDifference ?? 3;
+  const [isLoading, setIsLoading] = useState(false);
 
   const {
     control,
     formState: { errors },
     handleSubmit
   } = useForm({
     defaultValues: {
-      body: '',
-      dateEnd: new Date(),
-      dateStart: new Date(),
-      email: '',
-      name: '',
-      noticeboardType: '',
-      termsOfService: false,
-      title: ''
+      id: data?.id ?? '',
+      body: data?.contentBlocks?.[0]?.body ?? '',
+      dateEnd: data?.dates?.[0]?.dateEnd
+        ? moment(data?.dates?.[0]?.dateEnd)?.toDate()
+        : moment().add(requestedDateDifference, 'months').toDate(),
+      dateStart: data?.dates?.[0]?.dateStart
+        ? moment(data?.dates?.[0]?.dateStart)?.toDate()
+        : moment().toDate(),
+      email: data?.contacts?.[0]?.email ?? '',
+      image: existingImageUrl ? JSON.stringify([{ uri: existingImageUrl }]) : '[]',
```

Comment:
I get this error when I want to create a noticeboard.

![Simulator Screenshot - iPhone 15 Pro Max - 2024-07-29 at 09 29 06](https://github.com/user-attachments/assets/ee17e67e-eb22-4a6e-992b-140a20492d8f)